dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/airflow/AirflowConnector.java

```diff
@@ -95,24 +117,107 @@ public void addTasksTo(@Nonnull List<? super Task<?>> out, @Nonnull ConnectorArg
     out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
     out.add(new FormatTask(FORMAT_NAME));
 
+    ZonedIntervalIterable zonedIntervals = queryRangeFromConnectorArguments(arguments);
+    Pair<ZonedDateTime, ZonedDateTime> dateRange = dateRange(arguments);
+
     // Airflow v1.5.0
-    addFullTable(out, "dag.csv", "select * from dag;");
-    addFullTable(out, "task_instance.csv", "select * from task_instance;");
+    addQueryTask(out, "dag.csv", "select * from dag;");
+    // todo think about multiple files with this.addTableTaskWithIntervals()
+    addQueryTask(
+        out,
+        "task_instance.csv",
+        dateRange == null
+            ? "select * from task_instance;"
+            : "select * from task_instance "
+                + String.format(
+                    " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
+                    dateToSqlFormat(dateRange.getLeft()), dateToSqlFormat(dateRange.getRight())));
```

Comment:
If pre-build the filter query separately it might be easier to read and debug.
```suggestion
    String queryFilter = "";
    if (dateRange != null) {
      queryFilter = String.format(
           "where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
           dateToSqlFormat(dateRange.getLeft()),
           dateToSqlFormat(dateRange.getRight())
       )
    }
    addQueryTask(
        out,
        "task_instance.csv",
        String.format("select * from task_instance %s;", queryFilter)
```

Reply:
```
String filter =
        dateRange != null
            ? String.format(
                " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
                dateToSqlFormat(dateRange.getLeft()), dateToSqlFormat(dateRange.getRight()))
            : "";
addQueryTask(out, "task_instance.csv", "select * from task_instance " + filter);
```
vs
```
    addQueryTask(
        out,
        "task_instance.csv",
        "select * from task_instance "
            + (dateRange != null
                ? String.format(
                    " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
                    dateToSqlFormat(dateRange.getLeft()), dateToSqlFormat(dateRange.getRight()))
                : ""));

```
vs    
```
  addQueryTask(
        out,
        "task_instance.csv",
        dateRange == null
            ? "select * from task_instance;"
            : "select * from task_instance "
                + String.format(
                    " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
                    dateToSqlFormat(dateRange.getLeft()), dateToSqlFormat(dateRange.getRight())));
```

Not a big difference from my point of view. The variable looks good, but it will require to introduce such variable for  several method or use code blocks `{ ... }`.  

So, I will keep it as is.

Reply:
The main difference that you can log your filter or set a breakpoint for debugging purposes. With current implementation you have to do it in `addQueryTask` which is called multiple times. Hence it will be harder to identify required call.